p2p/swarmhandlers.go

```diff
@@ -36,7 +36,7 @@ func (s *swarmImpl) onRegisterNodeRequest(n node.RemoteNodeData) {
 	}
 
 	s.peers[n.ID()] = rn
-
+	s.peerMapMutex.Unlock()
```

Comment:
Since there are places where we only query the peers map you should consider using RWMutex
[https://golang.org/src/sync/rwmutex.go](url)